### PR TITLE
Refine time profile for amdllpc

### PR DESCRIPTION
**Pull request by linqun**
1. Support wildcard in windows, linux shell support wildcard in native, we need do nothing on linux.
2. Add dummy record to refine time-passes report
3. Output file name and pipeline hash when time-passes is enable

### CODE REVIEW

**Comment by amdrexu** [comment]
I will leave it pending for one day to see if anyone will have some comments.

File: tool/amdllpc.cpp [hunk 1]
```diff
@@ -1324,6 +1367,56 @@ static Result ProcessPipeline(
...
+// Finds all filenames which can match input file name
+void FindAllMatchFiles(
+    const std::string&        inFileName,     // [in] Input file name, include wildcard
+    std::vector<std::string>* pOutFileNames)  // [out] Output file names which can match input file names
+{
+    WIN32_FIND_DATAA data = {};
+
+    // Separate folder name
+    std::string folderName;
+    auto separatorIndex = inFileName.find_last_of("/\\");
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: separatorIndex -> separatorPos

File: tool/amdllpc.cpp [hunk 2]
```diff
@@ -1324,6 +1367,56 @@ static Result ProcessPipeline(
...
+
+    FindClose(pSearchHandle);
+    return;
+}
+#else
+
+// =====================================================================================================================
+// Finds all filenames which can match input file name
+void FindAllMatchFiles(
+    const std::string&        inFileName,     // [in] Input file name, include wildcard
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: inFileName -> inFile

File: tool/amdllpc.cpp [hunk 3]
```diff
@@ -1324,6 +1367,56 @@ static Result ProcessPipeline(
...
+    std::vector<std::string>* pOutFileNames)  // [out] Output file names which can match input file names
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: pOutFileNames -> pOutFiles

match input file names -> match input file name

File: tool/amdllpc.cpp [hunk 4]
```diff
@@ -1364,22 +1457,79 @@ int32_t main(
 
     if (IsPipelineInfoFile(InFiles[0]) || IsLlvmIrFile(InFiles[0]))
     {
+        uint32_t nextIndex = 0;
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: nextIndex -> nextFile

File: tool/amdllpc.cpp [hunk 5]
```diff
@@ -1364,22 +1457,79 @@ int32_t main(
...
         // The first input file is a pipeline file or LLVM IR file. Assume they all are, and compile each one
         // separately but in the same context.
         for (uint32_t i = 0; (i < InFiles.size()) && (result == Result::Success); ++i)
         {
-            result = ProcessPipeline(pCompiler, InFiles[i]);
+            if (InFiles[i].find_last_of("*?") != std::string::npos)
+            {
+                std::vector<std::string> matchedFiles;
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: matchedFiles -> matchFiles

File: tool/amdllpc.cpp [hunk 6]
```diff
@@ -1364,22 +1457,79 @@ int32_t main(
...
     else
     {
         // Otherwise, join all input files into the same pipeline.
-        SmallVector<std::string, 6> inFiles;
-        for (const auto &inFile : InFiles)
+        if ((InFiles.size() == 1) &&
+           ((InFiles[0].find_last_of("*?") != std::string::npos)))
         {
-            inFiles.push_back(inFile);
+            std::vector<std::string> matchedFiles;
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: matchedFiles -> matchFiles

File: tool/amdllpc.cpp [hunk 7]
```diff
@@ -1364,22 +1457,79 @@ int32_t main(
...
+            }
         }
     }
     else
     {
         // Otherwise, join all input files into the same pipeline.
-        SmallVector<std::string, 6> inFiles;
-        for (const auto &inFile : InFiles)
+        if ((InFiles.size() == 1) &&
+           ((InFiles[0].find_last_of("*?") != std::string::npos)))
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: One pair of () is redundant

File: tool/amdllpc.cpp [hunk 8]
```diff
@@ -1364,22 +1457,79 @@ int32_t main(
...
+            std::vector<std::string> matchedFiles;
+            FindAllMatchFiles(InFiles[0], &matchedFiles);
+            if (matchedFiles.size() == 0)
+            {
+                LLPC_ERRS("\nFailed to read file " << InFiles[0] << "\n");
+                result = Result::ErrorInvalidValue;
+            }
+            else
+            {
+                uint32_t nextIndex = 0;
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: nextIndex -> nextFile

File: tool/amdllpc.cpp [hunk 9]
```diff
@@ -1364,22 +1457,79 @@ int32_t main(
...
+                    LLPC_ERRS("\nCan't use wilecard if multiple filename is set in command\n");
+                    result = Result::ErrorInvalidValue;
+                    break;
+                }
+                inFiles.push_back(inFile);
+            }
+
+            if (result == Result::Success)
+            {
+                uint32_t nextIndex = 0;
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: nextIndex -> nextFile

**Comment by amdrexu** [comment]
Please click "resolve conversation" once you fix the issues.

File: tool/amdllpc.cpp [hunk 10]
```diff
@@ -1053,18 +1070,21 @@ static void EnableMemoryLeakDetection()
 // =====================================================================================================================
 // Process one pipeline.
 static Result ProcessPipeline(
-    ICompiler*            pCompiler,  // LLPC context
-    ArrayRef<std::string> inFiles)    // Input filename(s)
+    ICompiler*            pCompiler,   // LLPC context
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: // LLPC context -> // [in] LLPC context

File: tool/amdllpc.cpp [hunk 11]
```diff
@@ -1364,22 +1455,79 @@ int32_t main(
 
     if (IsPipelineInfoFile(InFiles[0]) || IsLlvmIrFile(InFiles[0]))
     {
+        uint32_t nextFile = 0;
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: Add a blank line after this line

File: tool/amdllpc.cpp [hunk 12]
```diff
@@ -1290,10 +1324,18 @@ static Result ProcessPipeline(
             result = CompileGlsl(inFile, &stage, spvBinFile);
             if (result == Result::Success)
             {
+                if (compileInfo.stageMask & ShaderStageToMask(static_cast<ShaderStage>(stage)))
+                {
+                    break;
+                }
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: Add a blank line after this line

File: tool/amdllpc.cpp [hunk 13]
```diff
@@ -1364,22 +1455,79 @@ int32_t main(
...
+                {
+                    result = ProcessPipeline(pCompiler, matchFiles[i], 0, &nextFile);
+                }
+            }
+        }
+        else
         {
-            inFiles.push_back(inFile);
+            SmallVector<std::string, 6> inFiles;
+            for (const auto &inFile : InFiles)
```
Comment (amdrexu) [inline tool/amdllpc.cpp]: const auto & -> const auto&

**Comment by amdrexu** [comment]
It is OK to merge.

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by JacobHeAMD** [comment]
retest this please

**Comment by JacobHeAMD** [comment]
ok to test

**Comment by JacobHeAMD** [comment]
test this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 44c3d105eaab5e794a946de40b098e320fcfc996
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev f40b53f526b1221a4efbd7beb00152ab32036882</li>
<li>SPVGEN: dev 1560d287f779b342e0019499dda85890cb07affc</li>
<li>XGL: dev 42e8b58860e4c049a3545d2559ba56420841dd1f</li>
<li>LLVM: amd-vulkan-dev 97cc33415120ae3ed472b6dd5cb234b74a80bd80</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 902/902 (100.0%)</li>
<li>Failed: 0/902 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.1.2-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 312622/312622 (100.0%)</li>
<li>Failed: 0/312622 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Vega20</summary><ul>

<li>Passed: 312687/312687 (100.0%)</li>
<li>Failed: 0/312687 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**